articles/nginx-reverse-proxy/index.md

```diff
@@ -0,0 +1,147 @@
+# Deploying Multiple Applications to VM with NGINX as a Reverse Proxy
+
+## Introduction
+Ever wondered how more than one application is being deployed to the same machine, and how traffic is being routed to the corresponding applications?
+Keep reading to know all about it!
+## Goals
+By the end of the article, one would be able to understand
+- What a Reverse Proxy is?
+- What NGINX is?
+- How does it help in managing multiple applications?
+- How to leverage NGINX as a Reverse Proxy?
+
+### Reverse Proxy
+A reverse proxy is a type of proxy server that retrieves resources on behalf of a client from one or more servers. These resources are then returned to the client, appearing as if they originated from the server itself.   
+
+### NGINX
+We will be using NGINX which would be serving the purpose of a Reverse Proxy. NGINX is a web server that can be used as a reverse proxy, load balancer, mail proxy and HTTP cache. 
+
+### Example
+![image](./reverse_proxy_working.png)
+
+This is an example of an architecture, where two apps are running in the background, but clients have no idea about them. Clients only know about NGINX which acts as a reverse proxy that sends the request to the appropriate application.
+
+
+Now that you have a broad idea of what we are about to build, let us jump right in!
+
+## Aim
+1. Deploy two applications and have it managed by NGINX.
+
+## Setup & Pre-Requisites:
+
+1. For this example, I have two sample NodeJS servers. One can have any kind of application running on different ports. 
+**NOTE:** Do not run your application on Port 80 or 443. Will be explained later why this must not be done.
+2. I have installed NGINX on my local machine, the same could be done on any Virtual Machine where the applications are expected to be deployed.
+
+## Step 1: Start two apps running in different ports
+As I've mentioned earlier, I got two Node Apps running on two different ports as shown below.  
+
+**Server app running on Port 3000**
+
+![image](./server1.png)  
+
+**Client app running on Port 3001**  
+
+![image](./client1.png)
+
+Now that we have our apps up and running, we don't want our users to use these applications by typing their PORTS explicitly, so we need to map it with something which is more human-readable. In this example, I will be using subdomains to distinguish between them. Again one is free to use whichever distinguishing element suitable as per their requirement.
```

Comment:
Suggest other distinguishing elements such as differing domain names.

Reply:
done and thanks